lib/core/test/integration/Test/Integration/Scenario/CLI/Server.hs

```diff
@@ -0,0 +1,63 @@
+module Test.Integration.Scenario.CLI.Server
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Launcher
+    ( Command (..), StdStream (..))
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe )
+import Control.Monad
+    ( when )
+import System.Directory
+    ( doesFileExist, removePathForcibly )
+import Test.Hspec
+    ( SpecWith, after_, describe, it )
+import Test.Integration.Framework.DSL
+    ( Context (..), expectCmdStarts )
+
+spec :: SpecWith (Context t)
+spec = after_ tearDown $ do
+    describe "SERVER - cardano-wallet server" $ do
+        it "SERVER - Can start cardano-wallet server without --database" $ \_ -> do
+            let cardanoWalletServer = Command
+                    "cardano-wallet"
```

Comment:
Note that we should use `stack -- exec cardano-wallet` and not `cardano-wallet` as the latter requires the software to have been installed. Which isn't usually the case when tests are ran, so it's really easy to just run them on a wrong version.